src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py

```diff
@@ -64,6 +94,46 @@ def __init__(self):
         for pin in range(4):
             self.gp_set_mode(pin, self.GP_GPIO)  # set to GPIO mode
             self.gpio_set_direction(pin, 1)  # set to INPUT
+        return self
+
+    @staticmethod
+    def available_paths(*, require_mcps=False):
+        """Return a list of paths of the currently available MCP2221s.
+
+        Raises a RuntimeError if *require_mcps* is True and there
+        are no MCP2221 connected.
+        """
+        mcps = hid.enumerate(MCP2221.VID, MCP2221.PID)
+        if require_mcps and not mcps:
+            raise RuntimeError('No MCP2221 detected.')
+        return [mcp["path"] for mcp in mcps]
+
+    @classmethod
+    def new_instance(cls):
+        """Find a new MCP2221, create an instance, and return it."""
+        self = super().__new__(cls)
+        self._hid = hid.device()
+        bus_ids = cls.available_paths(require_mcps=True)
+        print(f'{bus_ids=}')
+        # loop through all the available MCPs
+        for bus_id in bus_ids:
+            print(f'Opening {bus_id}...', end=' ')
+            try:
+                self._hid.open_path(bus_id)
+                self._bus_id = bus_id
+                # cache the new instance
+                cls.instances[bus_id] = self
+                print('[OK]')
+                return self
+            except (OSError, RuntimeError) as e:
+                print(f'[ERR]: {e}')
+                if len(bus_ids) == 1:
+                    raise  # we failed to open the only MCP
+                continue  # try opening the next one
+        else:
+            print('Failed to open all hid devices')
+            raise RuntimeError(f'Can not open any of the {len(bus_ids)} '
+                               f'connected MCP2221 devices.')
```

Comment:
To maintain backward compatibility, if there's only one MCP, the error is re-raised as is.  When there are multiple MCPs, a generic error is returned if none of them can be opened.  This might end up hiding some information on the cause of the errors though.

Reply:
This code is now integrated in `__new__`.  The same comment applies.